src/extension.ts

```diff
@@ -103,3 +152,110 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
 // tslint:disable-next-line:no-empty
 export function deactivateInternal(): void {
 }
+
+async function setupLocalProjectFolder(uri: vscode.Uri, filePath: string, token: string, account: AzureAccount): Promise<void> {
+    const requiredInputs: ILocalDevelopmentRequiredInputs = getAllRequiredInputs(uri.query);
+    const resourceId: string | null = requiredInputs.resourceId;
+    const defaultHostName: string | null = requiredInputs.defaultHostName;
+    const devContainerName: string | null = requiredInputs.devContainerName;
+    const language: string | null = requiredInputs.language;
+    const downloadAppContent: string | null = requiredInputs.downloadAppContent;
+
+    if (resourceId && defaultHostName && devContainerName && language && downloadAppContent) {
+        ext.context.globalState.update(LocalDevelopmentGlobalStates.projectLanguage, getProjectLanguageForLanguage(language));
+        const vsCodeFilePathUri: vscode.Uri = vscode.Uri.file(filePath);
+        const toBeDeletedFolderPathUri: vscode.Uri = vscode.Uri.joinPath(vsCodeFilePathUri, 'temp');
+
+        try {
+            const functionAppName: string = getNameFromId(resourceId, true);
+            const downloadFilePath: string = vscode.Uri.joinPath(toBeDeletedFolderPathUri, `${functionAppName}.zip`).fsPath;
+
+            vscode.window.showInformationMessage(localize('settingUpFunctionAppLocalProjInfoMessage', 'Setting up project for function app "${0}" with language "${1}".', functionAppName, language));
```

Comment:
I have an idea for the localize thing, to make it a little cleaner (I think)... but its up to you, as it serves no functional value...

how about defining localize getter methods at the top? something like:

```
const settingUpFunctionAppLocalProjInfoMessage = (functionAppName, language) =>  localize('settingUpFunctionAppLocalProjInfoMessage', 'Setting up project for function app "${0}" with language "${1}".', functionAppName, language);
const restartingVsCodeInfoMessage = () => localize('restartingVsCodeInfoMessage', 'Restarting VS Code with your function app project');
```

And usage would be something like:

```
vscode.window.showInformationMessage(settingUpFunctionAppLocalProjInfoMessage(functionAppName, language));
```
Or maybe even have in a separate static class or file.. but like that all the strings show up in one place??

Reply:
Honestly, I don't think it is needed as the entire extension code is doing it inline like we have right now. What you are suggesting makes sense but it should not be a one off for our scenario.